content/knowledge-codemagic/build-versioning.md

```diff
@@ -314,3 +314,133 @@ Provided you have exported your Google Play Console service account credentials
 
 
 Alternatively, you can add a custom [Pre-build script](/flutter-configuration/custom-scripts) and write the build number to a file, which will be read from your `android/app/build.gradle` during the build (See details [here](https://github.com/codemagic-ci-cd/android-versioning-example/tree/autoversioning_through_file)).
+
+
+
+## Firebase App Distribution latest build version
+
+Use [`firebase-app-distribution get-latest-build-version`](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/firebase-app-distribution/get-latest-build-version.md#get-latest-build-version) action from Codemagic CLI tools to get the latest build version from Firebase App Distribution.
+
+In order to do that, you need to provide Firebase access credentials by providing `FIREBASE_SERVICE_ACCOUNT_CREDENTIALS` as arguments to the action, as defined below.
+
+Additionally, you will need to provide project ID (e.g. `228333310124`) and application ID (e.g. `1:228333310124:ios:5e439e0d0231a788ac8f09`).
+
+
+#### Creating Firebase service account credentials
+
+You will need to set up a [Firebase service account](https://docs.codemagic.io/yaml-publishing/firebase-app-distribution/).
+
+#### Configuring the API access environment variables
+
+{{< tabpane >}}
+{{< tab header="codemagic.yaml" >}}
+{{<markdown>}}
+1. Open your Codemagic app settings, and go to the **Environment variables** tab.
+2. Enter the `FIREBASE_SERVICE_ACCOUNT_CREDENTIALS` as **_Variable name_**.
+3. Copy and paste the credentials content as **_Variable value_**.
+4. Enter the variable group name, e.g. `firebase_credentials`. Click the button to create the group.
+5. Make sure the **Secure** option is selected.
+6. Click the **Add** button to add the variable.
+
+7. Add the variable group to your `codemagic.yaml` file
+{{< highlight yaml "style=paraiso-dark">}}
+  environment:
+    groups:
+      - firebase_credentials
+{{< /highlight >}}
+{{</markdown>}}
+{{< /tab >}}
+
+
+{{% tab header="Flutter workflow editor"%}}
+
+Add the `FIREBASE_SERVICE_ACCOUNT_CREDENTIALS` environment variable to your Flutter project in **App settings > Environment variables** (See the details [here](../flutter-configuration/env-variables)).
+{{% /tab %}}
+{{< /tabpane >}}
+
+
+<br>
+{{<notebox>}}
+**Tip:** Alternatively, credentials can be specified as a command argument with the dedicated flag, see the details [here](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/firebase-app-distribution/get-latest-build-version.md#--credentials--cfirebase_service_account_credentials). In any case, it is advisable to save the service account credentials file to an environment variable so that it can be accessed during a build without committing it to version control. The environment variable will be a fallback for the missing value in the script.
+{{</notebox>}}
+
+
+#### Set the build version with `agvtool` for Apple projects
```

Comment:
A better word order would be:
Set the build version for iOS projects using `agvtool`